doc/manual/rl-next/nix-env-shell.md

```diff
@@ -0,0 +1,12 @@
+---
+synopsis: "`nix env shell` is the new `nix shell`, and `nix shell` remains an accepted alias"
+issues: 10504
+prs: 10807
+---
+
+This is part of an effort to bring more structure to the CLI subcommands.
+
+`nix env` will be about the process environment.
+Future commands may include `nix env run` and `nix env print-env`.
```

Comment:
We will need a deprecation phase of `nix print-dev-env` if we want to get rid off it.
It's used by *A LOT* of people: https://github.com/nix-community/nix-direnv/blob/1223f170d83efbe3a6f494f073b1d6b2d3dbae94/direnvrc#L300
https://github.com/direnv/direnv/blob/6a73cc2d7d13054829501df8756db04e79ff6a29/stdlib.sh#L1310

Reply:
Ok. Looks you make aliases anyway. I would expect given that direnv is also part of any Linux distribution we will still have the current version in some redhat in 10 years time.

Reply:
Yes, these commands are not going away - not even deprecated.
Aliases are cheap to support, and `nix shell` is genuinely useful as a shorthand.
`print-dev-env` we might eventually hide or something, but that's for a later PR.